### PR TITLE
Fix apply_top_k_top_p_triton called by non-cuda logits Tensor

### PR DESCRIPTION
**Pull request by xli**
Summary:
[#33538](https://github.com/vllm-project/vllm/pull/33538) add apply_top_k_top_p_triton function; inside apply_top_k_top_p_triton, there is `assert logits.is_cuda` to ensure input is cuda logits tensor, so we should guard the call with `logits.is_cuda`

### CODE REVIEW

**Review by WoosukKwon — APPROVED** [review]
@xli Thanks for the PR.

@njhill I actually think we can fix this by removing `assert logits.is_cuda` (plus others like `k.is_cuda`). WDYT?

**Review by houseroad — APPROVED** [review]

**Comment by xli** [comment]
> @xli Thanks for the PR.
> 
> @njhill I actually think we can fix this by removing `assert logits.is_cuda` (plus others like `k.is_cuda`). WDYT?

Removing assertion does not work in my local test with our internal hardware test, so I kept this diff simple.

**Comment by njhill** [comment]
> > @xli Thanks for the PR.
> > @njhill I actually think we can fix this by removing `assert logits.is_cuda` (plus others like `k.is_cuda`). WDYT?
> 
> Removing assertion does not work in my local test with our internal hardware test, so I kept this diff simple.

Thanks @xli and apologies for not catching this before. I agree we should remove the `is_cuda`s, could you give more information about the failure that you encountered? Was it related to `torch.cuda.get_device_properties`? Ideally we should fix the `HAS_TRITON` logic to exclude your case if needed rather than the additional check here.
